packages/deployments/contracts/contracts/messaging/connectors/wormhole/BaseWormhole.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.17;
+
+import {IWormholeRelayer} from "../../interfaces/ambs/wormhole/IWormholeRelayer.sol";
+import {IWormholeReceiver} from "../../interfaces/ambs/wormhole/IWormholeReceiver.sol";
+
+import {GasCap} from "../GasCap.sol";
+import "forge-std/console.sol";
+
+abstract contract BaseWormhole is GasCap, IWormholeReceiver {
+  // ============ Storage ============
+  address public immutable relayerAddress;
+  uint16 public immutable MIRROR_CHAIN_ID;
+  mapping(bytes32 => bool) public processedWhMessages;
```

Comment:
if we are supposed to protect against wormhole replay, we have to keep in mind that these connectors will be dropped in the future and we could lose that state.

if we don't plan on managing that, replay protection should be at a different level since its only protected against during the lifetime of the connector

Reply:
Obviously we can't protect replaying stuff just with `processedWhMessages`, in case of dropping and replacing spoke connector. Maybe we need this mapping on other layer, like `MerkleTreeManager`.

Reply:
to effectively protect against replays, yes. however, it doesn't make sense to put different mappings there. other types of protection we could add:

1. Enforce no duplicate leaves are able to be added -- this would have to be on both the `RootManager`'s `MerkleTree` as well as the `SpokeConnector`
2. Ensuring duplicate messages are not processable twice -- this check is pending for the larger messaging upgrade, and exist son the execution layer.

both (1) and (2) **significantly** increase the scope and risk of this upgrade. what happens if there are duplicate leaves in the `RootManager`'s tree? specifically with the pricing utilities

Reply:
okay chatted with Rahul -- there should be no problem if we lose this check and get transport replays post connector upgrades. we can leave it in to prevent excess fees